executeQuery(String sql) cannot be called on preparedStatement

### Search first

- [X] I searched and no similar issues were found

### Description

With PR #4928 and Release v4.25.0 an [issue](https://github.com/liquibase/liquibase/pull/4928/files#diff-e4e1307e2ceb21e940f02e3c2196f2b4783663b4e1142933171d8917549bfad8R527) was introduced by calling `executeQuery(selectStatement.toString())` on a prepared statement.

This causes Liquibase to fail with an "executeQuery(String sql) cannot be called on preparedStatement" exception on MariaDB instances

### Steps To Reproduce

Run Update command on MariaDB. e.g:

```JAVA
try (java.sql.Connection connection = DbConnect.get().getConnection()) {
    Scope.child(config, () -> {
        try {
            Database database = DatabaseFactory.getInstance().findCorrectDatabaseImplementation(new JdbcConnection(connection));

            log.info("DatabaseProductName: {}", database.getDatabaseProductName());

            CommandScope updateCommand = new CommandScope(UpdateCommandStep.COMMAND_NAME);
            updateCommand.addArgumentValue(UpdateCommandStep.CHANGELOG_FILE_ARG, changeLogFile);
            updateCommand.addArgumentValue(DbUrlConnectionCommandStep.DATABASE_ARG, database);
            updateCommand.addArgumentValue(ShowSummaryArgument.SHOW_SUMMARY, UpdateSummaryEnum.SUMMARY);
            updateCommand.execute();

        } catch (Exception e) {
            throw new RuntimeException("Failed to run Liquibase update", e);
        }
    });
} catch (Exception e) {
    log.error("Initialization failed", e);
    throw new RuntimeException(e);
}
```

### Expected/Desired Behavior

Replace   
`ResultSet columnSelectRS = statement.executeQuery(selectStatement.toString());`  
with. 
`ResultSet columnSelectRS = statement.executeQuery();`


### Liquibase Version

`>= 4.25`

### Database Vendor & Version

MaraDB

### Liquibase Integration

Java

### Liquibase Extensions

_No response_

### OS and/or Infrastructure Type/Provider

docker tomcat:9-jre17-temurin-jammy

### Additional Context

```LOG
Caused by: java.lang.reflect.InvocationTargetException
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
  at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]
  at org.apache.tomcat.jdbc.pool.StatementFacade$StatementProxy.invoke(StatementFacade.java:130) ~[tomcat-jdbc.jar:?]
  at jdk.proxy2/jdk.proxy2.$Proxy33.executeQuery(Unknown Source) ~[?:?]
  at liquibase.snapshot.JdbcDatabaseSnapshot$CachingDatabaseMetaData$GetColumnResultSetCache.determineActualDataTypes(JdbcDatabaseSnapshot.java:527) ~[liquibase-core-4.25.0.jar:?]
  at liquibase.snapshot.JdbcDatabaseSnapshot$CachingDatabaseMetaData$GetColumnResultSetCache.fastFetchQuery(JdbcDatabaseSnapshot.java:406) ~[liquibase-core-4.25.0.jar:?]
  at liquibase.snapshot.ResultSetCache$SingleResultSetExtractor.fastFetch(ResultSetCache.java:336) ~[liquibase-core-4.25.0.jar:?]
  at liquibase.snapshot.ResultSetCache.get(ResultSetCache.java:69) ~[liquibase-core-4.25.0.jar:?]
  at liquibase.snapshot.JdbcDatabaseSnapshot$CachingDatabaseMetaData.getColumns(JdbcDatabaseSnapshot.java:326) ~[liquibase-core-4.25.0.jar:?]
  at liquibase.snapshot.jvm.ColumnSnapshotGenerator.addTo(ColumnSnapshotGenerator.java:163) ~[liquibase-core-4.25.0.jar:?]
  at liquibase.snapshot.jvm.JdbcSnapshotGenerator.snapshot(JdbcSnapshotGenerator.java:78) ~[liquibase-core-4.25.0.jar:?]
  at liquibase.snapshot.SnapshotGeneratorChain.snapshot(SnapshotGeneratorChain.java:49) ~[liquibase-core-4.25.0.jar:?]
  at liquibase.snapshot.DatabaseSnapshot.include(DatabaseSnapshot.java:313) ~[liquibase-core-4.25.0.jar:?]
  at liquibase.snapshot.DatabaseSnapshot.init(DatabaseSnapshot.java:106) ~[liquibase-core-4.25.0.jar:?]
  at liquibase.snapshot.DatabaseSnapshot.<init>(DatabaseSnapshot.java:59) ~[liquibase-core-4.25.0.jar:?]
  at liquibase.snapshot.JdbcDatabaseSnapshot.<init>(JdbcDatabaseSnapshot.java:34) ~[liquibase-core-4.25.0.jar:?]
  at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:215) ~[liquibase-core-4.25.0.jar:?]
  at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:244) ~[liquibase-core-4.25.0.jar:?]
  at liquibase.snapshot.SnapshotGeneratorFactory.getDatabaseChangeLogTable(SnapshotGeneratorFactory.java:255) ~[liquibase-core-4.25.0.jar:?]
  at liquibase.changelog.StandardChangeLogHistoryService.init(StandardChangeLogHistoryService.java:111) ~[liquibase-core-4.25.0.jar:?]
  at liquibase.command.core.helpers.DatabaseChangelogCommandStep.checkLiquibaseTables(DatabaseChangelogCommandStep.java:141) ~[liquibase-core-4.25.0.jar:?]
  at liquibase.command.core.helpers.DatabaseChangelogCommandStep.run(DatabaseChangelogCommandStep.java:91) ~[liquibase-core-4.25.0.jar:?]
  at liquibase.command.CommandScope.execute(CommandScope.java:214) ~[liquibase-core-4.25.0.jar:?]
  at com.moers.erm4sn.common.web.servlet.LiquibaseServletListener.lambda$liquibaseSchemaUpdate$0(LiquibaseServletListener.java:91) ~[common-web-1.jar:?]
  ... 47 more
Caused by: java.sql.SQLSyntaxErrorException: (conn=12) executeQuery(String sql) cannot be called on preparedStatement
  at org.mariadb.jdbc.export.ExceptionFactory.createException(ExceptionFactory.java:289) ~[mariadb-java-client-3.3.3.jar:?]
  at org.mariadb.jdbc.export.ExceptionFactory.create(ExceptionFactory.java:343) ~[mariadb-java-client-3.3.3.jar:?]
  at org.mariadb.jdbc.BasePreparedStatement.executeQuery(BasePreparedStatement.java:217) ~[mariadb-java-client-3.3.3.jar:?]
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
  at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]
  at org.apache.tomcat.jdbc.pool.interceptor.AbstractQueryReport$StatementProxy.invoke(AbstractQueryReport.java:214) ~[tomcat-jdbc.jar:?]
  at jdk.proxy2/jdk.proxy2.$Proxy33.executeQuery(Unknown Source) ~[?:?]
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
  at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]
  at org.apache.tomcat.jdbc.pool.interceptor.AbstractQueryReport$StatementProxy.invoke(AbstractQueryReport.java:214) ~[tomcat-jdbc.jar:?]
  at jdk.proxy2/jdk.proxy2.$Proxy33.executeQuery(Unknown Source) ~[?:?]
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
```

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR (Thank you!)

--- a/liquibase-standard/src/main/java/liquibase/snapshot/JdbcDatabaseSnapshot.java
+++ b/liquibase-standard/src/main/java/liquibase/snapshot/JdbcDatabaseSnapshot.java
@@ -561,7 +561,7 @@ public class JdbcDatabaseSnapshot extends DatabaseSnapshot {
                     statement.setString(2, tableName);
                 }
                 try {
-                    ResultSet columnSelectRS = statement.executeQuery(selectStatement.toString());
+                    ResultSet columnSelectRS = statement.executeQuery();
                     //
                     // Iterate the result set from the query and match the rows
                     // to the rows that were returned by getColumns() in order

